CONTRIBUTING.md

```diff
@@ -1,24 +1,430 @@
+3. Sign the host app in the **Signing & Capabilities** tab
+4. Navigate to **Pods** in the **Project Manager**
+5. Add **Signing** to the **host app** and **unit test** targets
+6. Select the **Unit-unit** scheme
+7. **Run** it to build and test
+
+**Alternatively**, disable signing in each target:
+1. Go to **Build Settings** tab
+2. Click **+**
+3. Select **Add User-Defined Setting**
+4. Add `CODE_SIGNING_REQUIRED` setting with a value of `NO`
+</details>
+<br> <!-- This new line is for styling purposes. -->
+
+<!-- #### **Sample Apps** -->
+
+#### **Style Guide**
+This code in this repo is styled in accordance to `clang-format` conventions.
+
+#### Styling your code
+The [./scripts/style.sh] script makes it easy to style your
+code during development.
+<details>
+<summary>Using the <b>style.sh</b> script</summary>
+
+<!-- The above line is intentionally left blank. -->
+```
+# Usage:
+# ./scripts/style.sh [branch-name | filenames]
+#
+# With no arguments, formats all eligible files in the repo
+# Pass a branch name to format all eligible files changed since that branch
+# Pass a specific file or directory name to format just files found there
+#
+# Commonly
+# ./scripts/style.sh your_branch
+```
+</details>
+
+If your PR is failing CI due to style issues please use
+the style script accordingly. If the style script is not working, ensure you
+have installed the necessary code styling tools outlined in the
+[Getting Started](#getting-started) section.
+
+
+#### Apple development style guides and resources
+
+Refer to the following resources when writing Swift or Objective-C code.
+* Swift
+    * [Google's Swift Style Guide][google-swift-style]
+    * [Swift's API Design Guidelines][swift-api-design-guide]
+* Objective-C
+    * [Google's Objective-C Style Guide][google-objc-style]
+    * [Apple's Coding Guidelines for Cocoa][coding-guidelines-for-cocoa]
+
+#### **An example Git workflow**
+This is a general overview of what the Git workflow may look like, from start to
+finish, when contributing code to Firebase.
+The below snippet is purely for reference purposes and is used to demonstarate
+what the workflow may look like, from start to finish.
+<details>
+<summary>View the workflow</summary>
+
+<!-- The above line is intentionally left blank. -->
+For developers without write access, you'll need to create a fork of Firebase
+instead of a branch. Learn more about forking a repo [here][github-forks].
+```console
+# Update your local master
+git checkout master
+git pull
+
+# Create a development branch
+git checkout -b my_feature_or_bug_fix
+
+# Code, commit, repeat
+git commit -m "a helpful commit message"
+
+# Push your local branch to the remote
+git push --set-upstream origin my_feature_or_bug_fix
+
+# Open a pull request on github.com
+
+# Resolve review feedback on opened PR
+git commit -m "implemented suggestion"
+git push
+
+# Your PR can be merged once it has been approved by a project member 🎉
```

Comment:
Worth mentioning _who_ performs the merge (esp. for folks who have write access to the repo).